### PR TITLE
notify for rule runs

### PR DESCRIPTION
**Pull request by querdenker2k**
needed for metrics
e. g. 
`openhab_rule_runs_total{metric="openhab.core.metric.rules",openhab_core_metric="true",rule="Weather.weatherOverview",rulename="Weather overview",} 62.0`

### CODE REVIEW

**Comment by querdenker2k** [comment]
@seime can you tell my what's wrong with the EventPublisher injection?
I am getting an error: `Unresolved requirement: Import-Package: org.openhab.core.automation.internal`

**Comment by seime** [comment]
Not really sure. But it is referring to an internal package which probably isn't exported (a star import has been added; import org.openhab.core.automation.*; )

**Comment by querdenker2k** [comment]
Got it, imported another class from *.internal.* which i didn't recognized.

File: src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    /**
+     * Creates a rule updated event.
+     *
+     * @param rule the new rule.
+     * @param oldRule the rule that has been updated.
+     * @param source the source of the event.
+     * @return {@link RuleUpdatedEvent} instance.
+     */
+    public static RuleUpdatedEvent createRuleUpdatedEvent(Rule rule, Rule oldRule, String source) {
```
Comment (seime) [inline src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java:127]: Unused code?

File: src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    /**
+     * Creates a rule removed event.
+     *
+     * @param rule the rule for which this event is created.
+     * @param source the source of the event.
+     * @return {@link RuleRemovedEvent} instance.
+     */
+    public static RuleRemovedEvent createRuleRemovedEvent(Rule rule, String source) {
```
Comment (seime) [inline src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java:160]: Also unused

File: src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    /**
+     * Creates a rule added event.
+     *
+     * @param rule the rule for which this event is created.
+     * @param source the source of the event.
+     * @return {@link RuleAddedEvent} instance.
+     */
+    public static RuleAddedEvent createRuleAddedEvent(Rule rule, String source) {
```
Comment (seime) [inline src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java:174]: Also unused. Are we missing the code that was supposed to use this method, or is maybe this PR still WIP?

**Comment by querdenker2k** [comment]
sorry, class was just a copy paste from oh core. removed all unused code for jrule now.

File: src/main/java/org/openhab/automation/jrule/internal/module/JRuleRuleProvider.java [hunk 4]
```diff
@@ -20,11 +21,14 @@
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
-import org.openhab.core.automation.Rule;
-import org.openhab.core.automation.RuleProvider;
+import org.openhab.automation.jrule.rules.JRule;
+import org.openhab.core.automation.*;
```
Comment (seaside1) [inline src/main/java/org/openhab/automation/jrule/internal/module/JRuleRuleProvider.java:25]: Should avoid * import

File: src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+/**
+ * Cloned from openhab Core
+ *
+ * @author Robert Delbrück - Initial contribution
+ * @see org.openhab.core.automation.internal.RuleEventFactory
+ */
+@NonNullByDefault
+@Component(service = EventFactory.class, immediate = true)
+public class RuleEventFactory extends AbstractEventFactory {
```
Comment (seaside1) [inline src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java:37]: All  class names should follow pattern with prefix JRule
JRuleEventFactory

**Review by seime — APPROVED** [review]

**Pull request merged**